### DISCUSSION TITLE
why I cannot jump back using <c-o> in current buffer?

### DISCUSSION THREAD

**Discussion by iujakchu** [nvimdev/lspsaga.nvim#1246] [Q&A] [answered]
After going to definition using `lspsaga goto_definition`,I cannot jump back using `<c-o>`.
It's weird that I can only jump back when `goto_definition` jumps to another buffer.

**Comment by glepnir** [comment]
> After going to definition using lspsaga goto_definition,I cannot jump back using <c-o>.

on same buffer ?

**Reply by iujakchu** [reply]
Yes,is it expected behavior?

**Reply by glepnir** [reply]
maybe not . I will check it later.

**Reply by glepnir** [reply]
fixed update.

**Reply by iujakchu** [reply]
I updated to the latest version,but it seems no help.
`goto_definition` doesn't populate `jumps` .

**Reply by iujakchu** [reply]
Also,I confirmed it with this minimal config:
```lua
vim.loader.enable()
local plugins_path = vim.fn.stdpath("data") .. "/lazy"

local function boot(name, url)
	local package_path = plugins_path .. "/" .. name
	if not vim.loop.fs_stat(package_path) then
		vim.fn.system({
			"git",
			"clone",
			"--filter=blob:none",
			"--single-branch",
			url,
			package_path,
		})
	end
	vim.opt.runtimepath:prepend(package_path)
end

boot("lazy.nvim", "https://github.com/folke/lazy.nvim.git")
require("lazy").setup({
	{ "neovim/nvim-lspconfig", config = function()
		local lsp = require("lspconfig")
		lsp.pyright.setup{}
	end},
	{ "glepnir/lspsaga.nvim", config = true },
})


```
`nvim --clean` and `vim` and `:lua vim.lsp.buf.definition()` does not have this problem.

**Reply by glepnir** [reply, marked as answer]
fixed.
